cachi2/core/package_managers/pip.py

```diff
@@ -1323,9 +1321,9 @@ def from_line(cls, line: str, options: list[str]) -> Optional["PipRequirement"]:
 
         requirement.download_line = to_be_parsed
         requirement.options = options
-        requirement.package = req.project_name
+        requirement.package = _standardize_name(req.name)
```

Comment:
Not objecting to this line but I'm unclear whether packaging.Requirement already doesn't have the name attribute normalized so we could potentially just do req.name, but I admit I'm unsure on this one particularly.

Reply:
I ran a few tests and after changing from `pkg_resources.parse_requirements` to `Requirement` directly, we do need to ensure it's normalized.

Reply:
So, I dug a little deeper into this. To keep the previous behavior, we need to normalize here. 

Instantiating a req will allow the name to be anything that matches this [regex](https://github.com/pypa/packaging/blob/main/src/packaging/_tokenizer.py#L82). This means that `_` is allowed in the requirement line that is being parsed.

Previously, `project_name` was normalized using [safe_name](https://github.com/pypa/setuptools/blob/main/pkg_resources/__init__.py#L1550) from pkg_resources, which handles the `_` in the very least.

Reply:
This other [comment](https://github.com/containerbuildsystem/cachi2/pull/542#discussion_r1688793036) is relevant for here as well.

Reply:
> Not objecting to this line but I'm unclear whether packaging.Requirement already doesn't have the name attribute normalized so we could potentially just do req.name, but I admit I'm unsure on this one particularly.

The answer is NO, `packaging.Requirement` doesn't have the name normalized yet (but there is a TODO comment, so maybe one day :) ): https://github.com/pypa/packaging/blob/bb61b7bcbc429b49e6659a8b6c2966b6ebd046b6/src/packaging/requirements.py#L32